### PR TITLE
👷 Fix docker-compose integration tests

### PR DESCRIPTION
**Pull request by murgle2**

### CODE REVIEW

**Comment by Hona** [comment]
Please resolve the merge conflicts.

Also, it looks like the github actions aren't running on the latest commit here?

**Comment by murgle2** [comment]
> Please resolve the merge conflicts.
> 
> Also, it looks like the github actions aren't running on the latest commit here?

Resolved!

**Comment by Hona** [comment]
Is this ready for test?

**Comment by murgle2** [comment]
> 

Yep! It just needs that other PR (#561) so it will test the docker part correctly

**Comment by Gocnak** [comment]
Server tests aren't running right now. It's building the containers fine but the output is pretty much meaningless (if it's actually running tests, which considering GitHub actions moves on immediately, I don't think so):
image

**Review by Gocnak — CHANGES_REQUESTED** [review]
Server tests aren't running right now. It's building the containers fine but the output is pretty much meaningless (if it's actually running tests, which considering GitHub actions moves on immediately, I don't think so):
image

I would also prefer for the Docker containers to use a Yarn cache so we aren't waiting minutes again. 
image

This has been an issue since #559, I'm not sure what the rush was to get it in. @TheNerdMan also pointed out that there's no frontend dev (with auto-reloading) container either, which significantly hinders development progress.

**Comment by murgle2** [comment]
Server tests now shows tests outputs and correctly uses the exit code to determine test results. 

The benefit of dockerizing the backend is very significant in comparison to the frontend. The current dev workflow involves running yarn install and npm start for the frontend, which does autoreloading and communicates with the dockerized backend, which also autoreloads. 

If there's a strong demand for it, I can dockerize the frontend as well.

File: server/package.json [hunk 1]
```diff
@@ -4,8 +4,9 @@
   "private": true,
   "scripts": {
     "start": "node app.js",
+    "dev": "nodemon server.ts -L --ext ts,json,js",
     "ts-build": "tsc",
-    "test": "mocha --exit --timeout 45000"
+    "test": "chmod +x ../scripts/wait-for-it.sh && ../scripts/wait-for-it.sh db:3306 --timeout=60 -- mocha --exit --timeout 45000"
```
Comment (Gocnak) [inline server/package.json:9]: I think the `"test"` command should still be the `mocha --exit --timeout` stuff, and this specifically should be `test-docker`:

```suggestion
    "test": "mocha --exit --timeout 45000",
    "test-docker": "chmod +x ../scripts/wait-for-it.sh && ../scripts/wait-for-it.sh db:3306 --timeout=60 -- npm run test"
```

That way we're separating out the actual test command from the docker entrypoint command.

File: docker-compose.test.yml [hunk 2]
```diff
@@ -19,4 +19,4 @@ services:
     build:
       context: ./server
       dockerfile: dev.Dockerfile
-    entrypoint: sh -c "chmod +x ../scripts/wait-for-it.sh && ../scripts/wait-for-it.sh db:3306 --timeout=60 -- npm test"
+    entrypoint: "npm run test"
```
Comment (Gocnak) [inline docker-compose.test.yml:22]: And with the other suggestion:
```suggestion
    entrypoint: "npm run test-docker"
```

**Review by Gocnak — CHANGES_REQUESTED** [review]
Yeah the frontend just being a local dev environment is fine! The backend was the more tedious setup anyways, so great that it's properly containerized now.

I'd say caching is absolutely still worth it but can be done in another branch. A few nits here but I'm otherwise fine to merge this in after.

**Review by Gocnak — APPROVED** [review]

**Pull request merged**